docs/profiler.md

```diff
@@ -0,0 +1,112 @@
+
+## How to interpret the results
+
+```
+./profiler --data-size 1 --threads=1 --data-file data.bin --nonces=64
+{
+  "time_s": 12.09140029,
+  "speed_gib_s": 0.41351703525481415
+}
+```
+
+Here we see the following
+* 1GiB file `data.bin` was used for the benchmark
+* 1 thread was used
+* 64 nonces were used
+
+The benchmark took 12.09 seconds to complete and the speed was 0.41GiB/s.
+But from previous formula we know that the probability of finding a proof with 64 nonces is 79.39%. Therefore there is 20% chance that we will need to read the data again. Therefore the actual time can be significanly longer.
+```
+./profiler --data-size 1 --threads=1 --data-file data.bin --nonces=128
+{
+  "time_s": 13.152850458,
+  "speed_gib_s": 0.22808744078552953
+}
+```
+
+We see here that the speed dropped by half. That clearly indicates that CPU is our limiting factor.
+
+With 128 nonces there is 95.75% chance to find a proof in one data pass.
+
+```
+./profiler --data-size 1 --threads=10 --data-file data.bin --nonces=128
+{
+  "time_s": 10.331206291,
+  "speed_gib_s": 1.8390882405040923
+}
+```
+
+With 10 threads we can see that speed increased by 8 times. But again it's important to remember that we don't necessarily need to optimize for the speed, we need to optimize for the probability of finding a proof in time. There are 12 hours to find and submit proof.
+
+Based on these outputs you need to decide what is the best configuration for your hardware. Please note that the speed of the proof generation is not the only factor.
+
+
+
+## Tips & Hints
+
+### How to see that CPU is my limiting factor?
+If `speed_gib_s` is slowing down with more noces, then CPU is most likely the limiting factor. Try to add more threads IF needed
+
+### How to see that disk is my limiting factor?
+If `speed_gib_s` is not slowing down with more nonces, then disk is most likely the limiting factor. Try to add more nonces IF needed. That should put more load on CPU and limit the probability of reading the data again from the disk.
+
+### How do I find the sweet spot?
+That depends, if you want to generate proof as fast as possible because you cannot have your computer working for 12hours then make it work as hard as possible regardless of anything else.
+
+The general rule of thumb is that you want to match your disk speed with that benchmark. If your disk will be faster then CPU will be working on 100% and disk will NOT be fully utilized, if your CPU is way faster than HDD then CPU may be waiting for the data from the disk.
+
+### Do I need to finish the proof asap?
+No, you need to be in time, on the mainnet that time is 12hours. As long as you're on time you're good.
+
+
+## How to use the values
+
+```
+"smeshing-proving-opts": {
+  "smeshing-opts-proving-nonces": 144,
+  "smeshing-opts-proving-threads": 0
+},
+```
+
+Place that in your node config. Please note that the values are just an example and you need to use your own values.
```

Comment:
People will not know where to place it in the config. Consider specifying the whole path in JSON:
```
{
  "smeshing": {
    "smeshing-proving-opts": {
      "smeshing-opts-proving-nonces": 144,
      "smeshing-opts-proving-threads": 0
    }
  }
}
```

Reply:
Not sure honestly, that would assume that people know how to merge json. I gave flat part of a file that they have to add (including semicolon at the end)

Reply:
OK, but where to put it?

Reply:
hmm ok you convinced me. I'll add "smashing" also :) but I'll drop the top most {} ok?

Reply:
@poszu check now, changed my mind a bit and tried to make it more explicit.